src/Client.sol

```diff
@@ -0,0 +1,175 @@
+     * @dev This function can only be called by the manager
+     * @param client The address of the client for whom the allowed storage providers are being removed
+     * @param allowedSPs_ The list of storage providers to remove
+     */
+    function removeAllowedSPsForClient(address client, uint64[] calldata allowedSPs_) external onlyManager {
+        for (uint256 i = 0; i < allowedSPs_.length; i++) {
+            clientSPs[client][allowedSPs_[i]] = false;
+        }
+        emit SPsChanged(allowedSPs_);
+    }
+
+    /**
+     * @notice This function sets the manager address
+     * @dev This function can only be called by the manager
+     * @param manager_ The address of the manager
+     */
+    function setManager(address manager_) external onlyManager {
+        manager = manager_;
+        emit ManagerChanged(manager_);
+    }
+
+    /**
+     * @dev modifier to check if the caller is the manager
+     */
+    modifier onlyManager() {
+        if (msg.sender != manager) {
+            revert Errors.NotManager();
+        }
+        _;
+    }
+
+    /**
+     * @dev This function reads the array of AllocationRequests from the operator data and returns the providers
+     * @param cborData The cbor encoded operator data
+     * @param byteIdx The index of the byte to start reading from
+     */
+    function _deserializeAllocationRequests(bytes memory cborData, uint256 byteIdx)
+        internal
+        pure
+        returns (uint64[] memory providers)
+    {
+        uint256 operatorDataLength;
+        uint256 allocationRequestsLength;
+        uint64 provider;
+
+        (operatorDataLength, byteIdx) = CBORDecoder.readFixedArray(cborData, byteIdx);
+        if (operatorDataLength != 2) revert Errors.InvalidOperatorData();
+
+        (allocationRequestsLength, byteIdx) = CBORDecoder.readFixedArray(cborData, byteIdx);
+
+        providers = new uint64[](allocationRequestsLength);
+
+        for (uint256 i = 0; i < operatorDataLength; i++) {
+            uint256 allocationRequestLength;
+            (allocationRequestLength, byteIdx) = CBORDecoder.readFixedArray(cborData, byteIdx);
+
+            if (allocationRequestLength != 6) {
+                revert Errors.InvalidAllocationRequest();
+            }
+
+            (provider, byteIdx) = CBORDecoder.readUInt64(cborData, byteIdx);
+            providers[i] = provider;
+
+            (, byteIdx) = CBORDecoder.readBytes(cborData, byteIdx);
+            (, byteIdx) = CBORDecoder.readUInt64(cborData, byteIdx);
+            (, byteIdx) = CBORDecoder.readInt64(cborData, byteIdx);
+            (, byteIdx) = CBORDecoder.readInt64(cborData, byteIdx);
+            (, byteIdx) = CBORDecoder.readInt64(cborData, byteIdx);
+        }
+    }
+
+    /**
+     * @dev This function matches the providers with the allowedSPs
+     * @param providers The providers list to match
+     */
+    function _matchAllowedSPs(uint64[] memory providers) internal view {
```

Comment:
This name doesn't really indicate that it will revert. Can we rename it to something more like `_ensureSPsAreAllowed`?